### ISSUE TITLE
ocaml-findlib: error: configure: m4 not in PATH; this is required

### ISSUE THREAD

**Issue by ntysdd** [Homebrew/linuxbrew-core#12086]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Installing ocaml-findlib on Ubuntu
`brew install ocaml-findlib`

- What happened (include command output)
```
./configure
-bindir
/home/linuxbrew/.linuxbrew/Cellar/ocaml-findlib/1.8.0/bin
-mandir
/home/linuxbrew/.linuxbrew/Cellar/ocaml-findlib/1.8.0/share/man
-sitelib
/home/linuxbrew/.linuxbrew/Cellar/ocaml-findlib/1.8.0/lib/ocaml
-config
/home/linuxbrew/.linuxbrew/etc/findlib.conf
-no-topfind

Welcome to findlib version 1.8.0
Configuring core...
configure: m4 not in PATH; this is required
```
I installed the m4 formula, and it's in the PATH.
```
$ which m4
/home/linuxbrew/.linuxbrew/bin/m4
```

- What you expected to happen
no error

- Step-by-step reproduction instructions (by running `brew install` commands)
brew install ocaml-findlib

https://gist.github.com/0834d63118abc23fb84effe089ab7a9c

**Comment by iMichka** [comment]
Hi. Yes, we need to define that dependency in the formula. Could you try to edit ocaml-findlib.rb and add `depends on "m4" unless OS.mac?". If that works for you, would you like to open a pull request to fix this?

**Comment by ntysdd** [comment]
> Hi. Yes, we need to define that dependency in the formula. Could you try to edit ocaml-findlib.rb and add `depends on "m4" unless OS.mac?`. If that works for you, would you like to open a pull request to fix this?

I installed the m4 formula, and it still complains.

Is this expected?

**Comment by iMichka** [comment]
I tested the change locally, and it works: #12091
I'll build a bottle so you can use the binaries.

**Comment by ntysdd** [comment]
@iMichka OK.

**Comment by iMichka** [comment]
Fixed by #12091
